pkg/gateway/sig/v2.go

```diff
@@ -111,9 +111,10 @@ func (a *V2SigAuthenticator) Parse() (SigContext, error) {
 			return sigCtx, ErrHeaderMalformed
 		}
 		sigCtx.signature = sig
+		a.sigCtx = sigCtx
```

Comment:
If you made this change then you don't need `sigCtx` at the func scope.  Move it into the IF, I guess, and just create it at the very last moment as a literal.  I think that will be more readable than a zero-initialize var with a larger scope.